packages/core/wallet/src/register.ts

```diff
@@ -0,0 +1,54 @@
+import type {
+    Wallet,
+    WalletEventsWindow,
+    WindowRegisterWalletEvent as WindowRegisterWalletEventInterface,
+    WindowRegisterWalletEventCallback,
+} from '@wallet-standard/base';
+
+declare const window: WalletEventsWindow;
+
+export function registerWallet(wallet: Wallet): void {
+    const callback: WindowRegisterWalletEventCallback = ({ register }) => register(wallet);
+
+    try {
+        window.dispatchEvent(new WindowRegisterWalletEvent(callback));
+    } catch (error) {
+        console.error('wallet-standard:register-wallet event could not be dispatched\n', error);
+    }
+
+    try {
+        window.addEventListener('wallet-standard:app-ready', ({ detail: api }) => callback(api));
+    } catch (error) {
+        console.error('wallet-standard:app-ready event listener could not be added\n', error);
+    }
+}
+
+class WindowRegisterWalletEvent extends Event implements WindowRegisterWalletEventInterface {
```

Comment:
should we extend [CustomEvent](https://developer.mozilla.org/en-US/docs/Web/Events/Creating_and_triggering_events#adding_custom_data_%E2%80%93_customevent) given we're using `detail`?

Reply:
CustomEvent was originally used by causes errors with Next.js because it's not available on Node.js. However, the API was designed to be compatible with CustomEvent so someone could use it on the client if they want.

Reply:
https://github.com/wallet-standard/wallet-standard/pull/67/commits/183843207822d70a9a8c996292dfd508aa9e150b

Reply:
(in the wallet event, this shouldn't matter, but it does for the app and it's better if we can compile the libraries in any env)